pkg/types/azure/platform.go

```diff
@@ -0,0 +1,30 @@
+package azure
+
+import "strings"
+
+// Platform stores all the global configuration that all machinesets
+// use.
+type Platform struct {
+	// Region specifies the Azure region where the cluster will be created.
+	Region string `json:"region"`
+
+	// BaseDomainResourceGroupName specifies the resource group where the azure DNS zone for the base domain is found
+	BaseDomainResourceGroupName string `json:"BaseDomainResourceGroupName,omitempty"`
```

Comment:
This is the only quirk I think,
Azure can't identify the dns zone with just the name. so I need somewhere to store the dns zone resource group chosen. 
see : https://github.com/openshift/installer/blob/a95ce4126732191c1e43503b03ee4b3eee24d521/pkg/asset/installconfig/basedomain.go#L47-L55

if there is a better place, let me know, this does the trick for now.

Reply:
cc @crawford

Reply:
> if there is a better place, let me know, this does the trick for now.

line 55 is using the BaseDomainResourceGroupName property to store the group.
where should I store it?

Reply:
I first wanted to store the azure dns zone **ID** in `a.BaseDomain`, but you have validation verifying that the value is a well formed domain name, so that didn't work.

Reply:
This is where you store it https://github.com/openshift/installer/blob/976f49d9a323ea7006380031f94c7b496016fd92/pkg/asset/manifests/dns.go#L74-L83

Reply:
I think I understand now. We don't have the concept of `BaseDomain` in azure. the DNSZone represents both the `BaseDomain` and the `DNSZone`.
So my suggestion for azure, is to do the following:
- in `installconfig/basedomain.go`, do nothing
- in `manifests/dns.go` query for the dnszones, and set `installConfig.Config.BaseDomain` along with the `config.DNSZone` structure.   

Is that ok for you?

Reply:
>Is that ok for you?

`create install-config` prompts users for options like what basedomain you want to use. for AWS we query the AWS API and give users a list to choose from and uses that as `install-config.baseDomain`
for Azure we can do the same or just prompt user for the base domain.

irrespective of the choice you make above https://github.com/openshift/installer/blob/976f49d9a323ea7006380031f94c7b496016fd92/pkg/asset/manifests/dns.go#L74-L83 will need to perform the lookup for `install-config.baseDomain` to find the public dnszone's ID.

>We don't have the concept of BaseDomain in azure.

It is not concept of cloud.. it is concept for openshift 4 installer.

Reply:
the difference is that in Azure, we the basedomain is the same as the zone. in the install config, I show the available zones (or domain). when you choose one, that's your base domain AND your its id is the zoneID. I don't need to prompt twice.

if I prompt in the user basedomain.go, I have nowhere to store the zoneId. I need this zoneID as a terraform variable. if I store it in the DNSConfig object, how do I get it out into the TFVars?
I can get the manifests.DNS{} object through the dependencies, but then I need to unmarshall the file, is that what you suggest to do?
Can we have a call so I can walk you through it?

Reply:
> for Azure we can do the same or just prompt user for the base domain.

I'm _already_ doing this but that gives both the `basedomain` _and_  the `ZoneID` and I need to store this `ZoneID` somewhere for terraform. storing it in `DNSConfig` doesn't give me access to it when generating the `TFVars` AFAIK.

Reply:
I'll just simplify the question :
Where can I store the `DNSZoneID` for azure that I need to serialize to the azure `TFVars` ? 
I don't think that `&configv1.DNSZone` is the right place, as I can't access it when I call `TFVars(...)` for azure

Reply:
To summarize our call, we leave it as is for now, with `BaseDomainResourceGroupName` on the platform object, until the `installconfig` assumptions are more inline with the requirements for azure.
This will allow me to reuse these parameters for terraform `TFVars`, and to set the `&configv1.DNSZone` struct in the manifests for the ingress controller.